src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -128,11 +139,20 @@ public AdjustmentRule[] GetAdjustmentRules()
             // as the rules now is public, we should fill it properly so the caller doesn't have to know how we use it internally
             // and can use it as it is used in Windows
 
-            AdjustmentRule[] rules = new AdjustmentRule[_adjustmentRules.Length];
+            List<AdjustmentRule> rulesList = new List<AdjustmentRule>(_adjustmentRules.Length);
 
             for (int i = 0; i < _adjustmentRules.Length; i++)
             {
                 AdjustmentRule? rule = _adjustmentRules[i];
+
+                if (rule.NoDaylightTransitions &&
```

Comment:
Should you check for `null`? rule is nullable here.

Reply:
I think we can make the rule not nullable. good catch.